### PR TITLE
@orta=> Presentation mode optimizations

### LINKED ISSUE #144: Optimizations for Presentation mode to avoid having it on but with no settings selected.
Not a release stopper, but depending on how big of an undertaking this is would be good, but not necessary to get in this version. Discussed with @lucaderosso today that the current designs provide the not ideal opportunity to have presentation mode toggled on but no presentation mode settings toggled on—creating an initially confusing user experience where you feel like you've turned something on but nothing has changed.

Possibility given current implementation:
img_0072

There are a couple optimizations (as well as minor related copy changes otherwise) we can make to minimize confusion by eliminating the possibility to have presentation mode on but with no settings changed.

- [x] the default settings on first sync of the app (or maybe also first sync with this settings design if possible) should have by-default all options under presentation mode toggled-ON. This way toggling presentation mode on will initiate max possible privacy view.

if all settings are toggled off by the user:
- [x] the primary callout for "Presentation Mode" should be grayed out with the toggle disabled,
- [x] Copy edit: "Edit Presentation Mode" should be "Presentation Mode Settings"
- [x] Copy edit: when presentation mode is disabled helper copy should read "Presentation Mode hides sensitive information when showing artworks to clients. To use presentation mode, turn on one or more options in Presentation Mode Settings."
- [x] Copy edit: helper copy for even if pres mode is enabled should change from "Hides sensitive info..." to "Presentation Mode hides sensitive information when showing artworks to clients."
artboard 1

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] Fixes blurry wifi icon images
- [x] Settings menu view model now initializes presentation mode
- [x] Better tests
- [x] Updated copy

Fixes #144

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Error
    ------------- | ------------
        :no_entry_sign: | No CHANGELOG changes made
    

  

  

  


<p align="right" meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

File: ArtsyFolio Tests/View Controllers/ARLabSettingsMasterViewControllerTests.m [hunk 1]
```diff
@@ -44,18 +51,37 @@ - (IBAction)presentationModeButtonPressed:(id)sender;
...
+        subject.viewModel = viewModelWithDefaultsAndContext(@{ ARHasInitializedPresentationMode: @YES, ARPresentationModeOn: @NO, ARHideArtworkEditButton: @YES }, context);
+        
+        expect(subject).to.haveValidSnapshot();
+    });
 });
 
 SpecEnd
+
+    ARLabSettingsMenuViewModel *
+    viewModelWithDefaultsAndContext(NSDictionary *defaults, NSManagedObjectContext *context)
```
Comment (orta) [inline ArtsyFolio Tests/View Controllers/ARLabSettingsMasterViewControllerTests.m]: this is great

File: Classes/Controllers/Settings/ARLabSettings.storyboard [hunk 2]
```diff
@@ -164,7 +164,7 @@
                                         <rect key="frame" x="0.0" y="30" width="558" height="44"/>
                                         <autoresizingMask key="autoresizingMask"/>
                                         <tableViewCellContentView key="contentView" opaque="NO" clipsSubviews="YES" multipleTouchEnabled="YES" contentMode="center" tableViewCell="liS-em-7gK" id="JDf-3k-llm">
-                                            <rect key="frame" x="0.0" y="0.0" width="558" height="43"/>
+                                            <rect key="frame" x="0.0" y="0.0" width="558" height="43.5"/>
```
Comment (orta) [inline Classes/Controllers/Settings/ARLabSettings.storyboard]: interesting, are you sure you want these?

File: Classes/Controllers/Settings/ARLabSettingsPresentationModeViewController.m [hunk 3]
```diff
@@ -102,7 +100,16 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
 
     ARToggleSwitch *toggle = [ARToggleSwitch buttonWithFrame:CGRectMake(0, 0, 76, 28)];
+
+    /// By default, all presentation mode options are OFF
+    //    if (!self.didInitializeOptions) {
+    //        toggle.on = YES;
+    //        [self.defaults setBool:YES forKey:option[AROptionsKey]];
+    //    } else {
+    //        toggle.on = [self.defaults boolForKey:option[AROptionsKey]];
+    //    }
```
Comment (orta) [inline Classes/Controllers/Settings/ARLabSettingsPresentationModeViewController.m]: ^

**Comment by orta** [comment]
Cool this feels good, not sure about those comments but everything else seems fine

**Comment by sarahscott** [comment]
fixed the comments; merging!

**Pull request merged**